Author: vlendec
Revision: r24649
Modified: /branches/SAMBA_3_0_25/source/nsswitch/idmap_ldap.c source/nsswitch/idmap_ldap.c /branches/SAMBA_3_2_0/source/nsswitch/idmap_ldap.c
Added: 
Removed: 


Attempt to fix bug 4917. Simo, please check!

Thanks Patrick Rynhart for reporting this.


Index: source/nsswitch/idmap_ldap.c
===================================================================
--- source/nsswitch/idmap_ldap.c	(revision 24648)
+++ source/nsswitch/idmap_ldap.c	(revision 24649)
@@ -896,7 +896,7 @@
 	uidNumber = get_attr_key2string(idpool_attr_list, LDAP_ATTR_UIDNUMBER);
 	gidNumber = get_attr_key2string(idpool_attr_list, LDAP_ATTR_GIDNUMBER);
 
-	attr_list = get_attr_list(ctx, sidmap_attr_list);
+	attr_list = get_attr_list(memctx, sidmap_attr_list);
 
 	if ( ! ids[1]) {
 		/* if we are requested just one mapping use the simple filter */
@@ -1113,7 +1113,7 @@
 	uidNumber = get_attr_key2string(idpool_attr_list, LDAP_ATTR_UIDNUMBER);
 	gidNumber = get_attr_key2string(idpool_attr_list, LDAP_ATTR_GIDNUMBER);
 
-	attr_list = get_attr_list(ctx, sidmap_attr_list);
+	attr_list = get_attr_list(memctx, sidmap_attr_list);
 
 	if ( ! ids[1]) {
 		/* if we are requested just one mapping use the simple filter */

